enclave-runner/src/tcs.rs

```diff
@@ -184,10 +184,12 @@ pub(crate) fn coenter<T: Tcs>(
         } else {
             asm!("
                     lea 1f(%rip), %rcx // set SGX AEP
+                    xchg {0}, %rbx
 1:                  enclu
+                    xchg %rbx, {0}
```

Comment:
Looks like `ENCLU[EENTER]` doesn't write to `rbx` so the second `xchg` can perhaps be reduced to `movq {0}, %rbx`?

https://www.felixcloutier.com/x86/eenter

I'm just trying to learn how different projects dealt with https://github.com/rust-lang/rust/pull/84658 and came across with this

Reply:
EEXIT and AEX do write to RBX. But we were already ignoring the value, so `mov` would've been ok in this case. However, `xchg` is more “obviously correct” (and revertible) to deal with this LLVM nonsense.